docs/NativeModulesIOS.md

```diff
@@ -173,6 +173,37 @@ A native module is supposed to invoke its callback only once. It can, however, s
 
 If you want to pass error-like objects to JavaScript, use `RCTMakeError` from [`RCTUtils.h`](https://github.com/facebook/react-native/blob/master/React/Base/RCTUtils.h).  Right now this just passes an Error-shaped dictionary to JavaScript, but we would like to automatically generate real JavaScript `Error` objects in the future.
 
+## Promises
+
+Native modules can also return a Promise, which can simplify your code quite lot, especially when using ES7 `async/await` syntax.
+
+If we refactor the above code using callbacks to use Promise, it becomes the following,
+
+```objective-c
+RCT_EXPORT_METHOD(resolver:(RCTPromiseResolveBlock)resolve
```

Comment:
Need to name this `RCT_EXPORT_METHOD(findEvents:(RCTPromiseResolveBlock)resolve` or else it will define a JS method called 'resolve' :)